pkg/app/api/grpcapi/web_api.go

```diff
@@ -1350,14 +1355,18 @@ func (a *WebAPI) GetInsightData(ctx context.Context, req *webservice.GetInsightD
 	count := int(req.DataPointCount)
 	from := time.Unix(req.RangeFrom, 0)
 
-	// TODO: caching the insight data inside the cache service
-	chunks, err := a.insightstore.LoadChunks(ctx, claims.Role.ProjectId, req.ApplicationId, req.MetricsKind, req.Step, from, count)
+	var chunks insight.Chunks
+	chunks, err = a.insightstore.LoadChunksCache(a.insightCache, claims.Role.ProjectId, req.ApplicationId, req.MetricsKind, req.Step, from, count)
 	if err != nil {
-		a.logger.Error("failed to load chunks from insightstore", zap.Error(err))
-		return nil, err
```

Comment:
Please add logs to show why `LoadChunksFromCache` and `PutChunksToCache` failed.